src/DFGNode.cpp

```diff
@@ -243,12 +284,67 @@ bool DFGNode::isLogic() {
   return false;
 }
 
-bool DFGNode::hasCombined() {
+bool DFGNode::isLut() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and op.compare("lut") == 0)
+    return true;
+  return false;
+}
+
+bool DFGNode::isDequantize() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and op.compare("Dequantize") == 0)
+    return true;
+  return false;
+}
+
+// Convert the fp8/fp16 to fp32 or int8/int16 to int32.
+bool DFGNode::isConvert() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and (op.compare("fpConvert") == 0 or op.compare("intConvert") == 0))
+    return true;
+  return false;
+}
+
+// Divison can also be a special operation.
+bool DFGNode::isDiv() {
+  if (m_opcodeName.compare("fdiv") == 0 or m_opcodeName.compare("div") == 0)
+    return true;
+  return false;
+}
+
+// Quantize the fp32 to fp8/fp16 or int32 to int8/int16.
+bool DFGNode::isQuantize() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and (op.compare("fpQuantize") == 0 or op.compare("intQuantize") == 0))
+    return true;
+  return false;
+}
+
+// used for specialized fusion (e.g. alu+mul and icmp+br can be regared as two kinds of complex nodes, so there are different tiles to support them)
+// type indicates the type of the combined node. (e.g. 0 for alu+mul, 1 for icmp+br)
+// type=-1 means the node is combined with any type, which is used for general fusion.
```

Comment:
`type=-1` -> `type = -1`.

And the `type = -1` case needs to be explained somewhere (or here). What do you mean by general fusion.

Reply:
I added some explanations here. And To make it easier to understand, I change the type of `type` from integer to string. Now `type` can be `BrT`, `CoT` and so on, like we used in PICACHU.